app/services/import/csv/activity_actual_refund_comment/row_service.rb

```diff
@@ -0,0 +1,135 @@
+class Import::Csv::ActivityActualRefundComment::RowService
+  attr_reader :errors
+
+  def initialize(report, user, row)
+    @report = report
+    @user = user
+    @activity = nil
+    @errors = {}
+    @row = Import::Csv::ActivityActualRefundComment::Row.new(row)
+  end
+
+  def import!
+    if @row.valid?
+      @activity = find_activity(@row.roda_identifier)
+
+      return false if @activity.nil?
+
+      return false unless authorise_activity(@activity)
+
+      return Import::Csv::ActivityActualRefundComment::SkippedRow.new(@row) if @row.empty?
+
+      create_record
+    else
+      @errors.update(@row.errors)
+      false
+    end
+  end
+
+  private def create_record
+    case type_of_row(@row)
+    when :actual
+      result = create_actual(@row, @activity)
+    when :refund
+      result = create_refund(@row, @activity)
+    when :comment
+      result = create_activity_comment(@row, @activity)
+    end
+
+    return false unless result
+
+    if result.failure?
+      result.object.errors.each do |error|
```

Comment:
I'm a little confused about what's happening here, with copying the errors in a slightly different format.

Reply:
These are the actual model errors which have a slightly different format - it is unlikely there will be any as we have already validate everything, but if at a later date the underlying model validation changes, this would surface the errors.

This is how the current implementation works:

https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/4f0a9d46d4a457206c29292caaf1b3a26e28f609/app/services/actual/import.rb#L42